htlcswitch/switch.go

```diff
@@ -1990,6 +1990,19 @@ func (s *Switch) getLinkByShortID(chanID lnwire.ShortChannelID) (ChannelLink, er
 	return link, nil
 }
 
+// ActiveLinks returns a list of ChannelLinks currently active within the
+// switch.
+func (s *Switch) ActiveLinks() []ChannelLink {
+	s.indexMtx.RLock()
+	defer s.indexMtx.RUnlock()
+
+	var links []ChannelLink
+	for _, l := range s.linkIndex {
+		links = append(links, l)
```

Comment:
should maybe filter by `EligibleToForward()`?

Reply:
Ah, good idea! Should the same filtering be done for `HasActiveLink` below?